dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/NettyChain.java

```diff
@@ -116,16 +113,23 @@ public synchronized void stop() {
             } finally {
                 if (previousState.val > ChainState.QUIESCED.val) {
                     VirtualHostMap.notifyStopped(owner, currentConfig.getResolvedHost(), currentConfig.getConfigPort(), isHttps);
-                    MSP.log("stop()-> VHOST notified");
                     currentConfig.clearActivePort();
                     String topic = owner.getEventTopic() + HttpServiceConstants.ENDPOINT_STOPPED;
                     postEvent(topic, currentConfig, null);
                 }
-                state.set(ChainState.STOPPED);
+                
+                //TODO: do we really need to do events when unitialized? 
+                if (previousState != ChainState.UNINITIALIZED) {
+                    state.set(ChainState.STOPPED);
+                } else {
+                    state.set(ChainState.UNINITIALIZED);
```

Comment:
The else does not look to be necessary.

Reply:
It becomes necessary due to line 95 `ChainState previousState = state.getAndSet(ChainState.STOPPING);`